File: src/main/java/gargoyle/netsquares/store/prop/PrefProperty.java
package gargoyle.netsquares.store.prop;

import gargoyle.netsquares.util.SerializationUtils;

import java.io.Serializable;

public class PrefProperty<T extends Serializable> extends BasePrefProperty<T> {
    public PrefProperty(Class<?> pkg, String key, T defaultValue) {
        super(pkg, key, defaultValue);
    }

    @Override
    protected void _set(T value) {
        preferences.putByteArray(key, SerializationUtils.serialize(value));
    }

    public T get(T defaultValue) {
        byte[] bytes = preferences.getByteArray(key, null);
        return bytes == null ? defaultValue : SerializationUtils.deserialize(bytes);
    }
}

File: src/main/kotlin/gargoyle/netsquares/store/prop/PrefProperty.kt
package gargoyle.netsquares.store.prop

import gargoyle.netsquares.util.SerializationUtils
import kotlin.reflect.KClass

class PrefProperty<T : Any>(pkg: KClass<*>, key: String, defaultValue: T) :
    BasePrefProperty<T>(pkg, key, defaultValue) {
    override fun _set(value: T) {
        preferences.putByteArray(key, SerializationUtils.serialize(value))
    }

    override fun get(defaultValue: T): T {
        val bytes = preferences.getByteArray(key, null)
        return if (bytes == null) defaultValue else SerializationUtils.deserialize(bytes)
    }
}
